src/python/grpcio/grpc/experimental/aio/_interceptor.py

```diff
@@ -325,14 +308,193 @@ async def debug_error_string(self) -> Optional[str]:
+        self._call_response_iterator = CallResponseIterator()
+        self._intercepted_response_iterator = None
+        interceptors_task = loop.create_task(
+            self._invoke(interceptors, method, timeout, metadata,
+                         credentials, wait_for_ready, request,
+                         request_serializer, response_deserializer)
+        )
+        super().__init__(interceptors_task)
+
+    # pylint: disable=too-many-arguments
+    async def _invoke(self,
+                      interceptors: Sequence[UnaryUnaryClientInterceptor],
+                      method: bytes, timeout: Optional[float],
+                      metadata: Optional[MetadataType],
+                      credentials: Optional[grpc.CallCredentials],
+                      wait_for_ready: Optional[bool], request: RequestType,
+                      request_serializer: SerializingFunction,
+                      response_deserializer: DeserializingFunction
+                     ) -> UnaryStreamCall:
+        """Run the RPC call wrapped in interceptors"""
+
+        async def _run_interceptor(
+                interceptors: Iterator[UnaryStreamClientInterceptor],
+                client_call_details: ClientCallDetails,
+                request: RequestType,
+                response_iterator: AsyncIterable[ResponseType]
+               ) -> _base_call.UnaryUnaryCall:
+
+            interceptor = next(interceptors, None)
+
+            if interceptor:
+                continuation = functools.partial(_run_interceptor, interceptors)
+
+                return await interceptor.intercept_unary_stream(
+                    continuation, client_call_details, request, response_iterator)
+            else:
+                call = UnaryStreamCall(
+                    request, _timeout_to_deadline(client_call_details.timeout),
+                    client_call_details.metadata,
+                    client_call_details.credentials,
+                    client_call_details.wait_for_ready, self._channel,
+                    client_call_details.method, request_serializer,
+                    response_deserializer, self._loop)
+
+                self._call_response_iterator.set_call(call)
+                self._intercepted_response_iterator = response_iterator
+                return call
+
+        client_call_details = ClientCallDetails(method, timeout, metadata,
+                                                credentials, wait_for_ready)
+        return await _run_interceptor(iter(interceptors), client_call_details,
+                                      request, self._call_response_iterator)
+
+    async def _wait_for_interceptor_task_response_iterator(self) -> ResponseType:
+        await self._interceptors_task
+        async for response in self._intercepted_response_iterator:
+            yield response
+
+    def __aiter__(self) -> AsyncIterable[ResponseType]:
+        return self._response_aiter
```

Comment:
I was wondering why this works, until I found the wrapper of async-generator is another async-generator. 👍